k2/csrc/CMakeLists.txt

```diff
@@ -6,30 +6,38 @@ add_subdirectory(host)
 
 #---------------------------- Build K2 CUDA sources ----------------------------
 
+# to make code static analysis work
+# @todo delete this include line before release
+include_directories(${CUDAToolkit_INCLUDE_DIRS})
```

Comment:
just to make sure, Is this path valid for Cmake with version < 3.18?

Reply:
Do we need to specify the include path explicitly? I think nvcc has its default searching paths that work well enough.

Reply:
Yeah, deleted it here and other places. 

In case of wondering the reason:
It's unnecessary if the sources with cuda keyword also with "cu[h]" suffix. In our coding philosophy, cmake would give the sources with cuda keyword to gcc, then gcc give errors. I would delete it now, and try what will happen without it.